mars/dataframe/execution/datasource.py

```diff
@@ -14,14 +14,26 @@
 
 from ..expressions.datasource.dataframe import DataFrameDataSource
 from ..expressions.datasource.series import SeriesDataSource
+from ..expressions.datasource.from_tensor import DataFrameFromTensor
+
+try:
+    import pandas as pd
+except ImportError:  # pragma: no cover
+    pass
 
 
 def _dataframe_or_series_pandas_data_source(ctx, chunk):
     ctx[chunk.key] = chunk.op.data
 
 
+def _dataframe_tensor_data_source(ctx, chunk):
+    tensor_data = ctx[chunk.inputs[0].key]
+    ctx[chunk.key] = pd.DataFrame(tensor_data, index=chunk.index_value, columns=chunk.columns)
```

Comment:
As I commented, `chunk.index_value` and `chunk.columns` is instances of `IndexValue`, so here, you should convert them to pandas via `to_pandas`, before passing them to `pd.DataFrame`.